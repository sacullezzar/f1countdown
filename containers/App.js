import React, { Component } from 'react'
import Timer from '../components/timer'
import RaceData from '../components/raceData'
import 'babel-polyfill'

class App extends Component {
    constructor(props) {
        super(props)
        this.state = {
            isLoading: true,
            number: 0,
            raceData: null
        }

        this.handleTimeChange = this.handleTimeChange.bind(this)
    }

    async componentDidMount() {
        const response = await fetch('https://ergast.com/api/f1/current.json')
        const responseJson = await response.json()
        this.setState({ isLoading: false, raceData: responseJson })
    }

    handleTimeChange(time) {
        const t1 = new Date
        const t2 = new Date(time)
        const date = Math.round((t2 - t1)/1000)
        this.setState({ number: date})
    }

    render () {
        const { number, raceData } = this.state
        return (
            <React.Fragment>
                <h1 style={{textAlign: "center"}}>F1 Countdown</h1>
                <Timer time={number}/>
                {!this.state.isLoading && <RaceData raceData={raceData} handleTimeChange={this.handleTimeChange}/>}
            </React.Fragment>
        )
    }
}

export default App