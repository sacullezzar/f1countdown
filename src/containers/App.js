import React, { Component } from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link
} from 'react-router-dom'
import Countdown from '../containers/Countdown'
import RaceSelect from '../components/raceSelect'
import DriversChamp from './DriversChamp'
import TeamChamp from './TeamChamp'

class App extends Component {
    constructor() {
        super()
        this.state = {
            isLoading: true,
            seasonData: null
        }
        this.fetchData = this.fetchData.bind(this)
    }

    componentDidMount() {
        this.fetchData()
    }

    async fetchData() {
        const season = await fetch('https://ergast.com/api/f1/current.json')
        const seasonJson = await season.json()
        this.setState({ isLoading: true, seasonData: seasonJson.MRData.RaceTable.Races })
        const standings = await fetch('https://ergast.com/api/f1/current/driverStandings.json')
        const standingsJson = await standings.json()
        this.setState({ isLoading: true, standingsData: standingsJson.MRData.StandingsTable.StandingsLists })
        const teamStandings = await fetch('https://ergast.com/api/f1/current/constructorStandings.json')
        const teamStandingsJson = await teamStandings.json()
        this.setState({ isLoading: false, teamData: teamStandingsJson.MRData.StandingsTable.StandingsLists})
    }

    render() {
        const { seasonData, standingsData, teamData, isLoading } = this.state
        return (
            <Router>
            <div>
            <nav>
                <ul>
                <li>
                    <Link to="/">Countdown</Link>
                </li>
                <li>
                    <Link to="/races">Previous Races</Link>
                </li>
                <li>
                    <Link to="/driversStandings">Driver's Championship Standings</Link>
                </li>
                <li>
                    <Link to="/constructorsStandings">Constructor's Championship Standings</Link>
                </li>
                </ul>
            </nav>
    
            {/* A <Switch> looks through its children <Route>s and
                renders the first one that matches the current URL. */}
            <Switch>
                <Route path="/races">
                    {seasonData && seasonData.length ? <RaceSelect  seasonData={seasonData}/> : <h2>Loading...</h2>}
                </Route>
                <Route path="/driversStandings">
                    {standingsData && standingsData.length ? <DriversChamp  standingsData={standingsData}/> : <h2>Loading...</h2>}
                </Route>
                <Route path="/constructorsStandings">
                    {teamData && teamData.length ? <TeamChamp teamData={teamData}/> : <h2>Loading...</h2>}
                </Route>
                <Route path="/">
                    {seasonData && seasonData.length ? <Countdown seasonData={seasonData} isLoading={isLoading}/> : <h2>Loading...</h2>}
                </Route>
            </Switch>
            </div>
        </Router>
        )
    }
}

export default App