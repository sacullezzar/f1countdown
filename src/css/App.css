.container {
    display: flex;
}

.title-box {
    width: 'auto';
}
.title {
    font-size: 4em;
    margin: 10,20,30,40;
    background-color: blueviolet;
}