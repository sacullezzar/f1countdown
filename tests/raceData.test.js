import React from 'react'
import RaceData from '../components/raceData'
import raceDataProp from './testData/raceData'
import App from '../containers/App'

describe('<raceData />', function () {
    it('first test', function() {
        const wrapper = shallow(<RaceData raceData={raceDataProp}/>)
        expect(wrapper).toMatchSnapshot()
    })

    it('expects the props function to have been called on select change', function() {
        const mockFn = jest.fn()
        const wrapper = shallow(<RaceData raceData={raceDataProp} handleTimeChange={mockFn}/>)
        wrapper.find('select').simulate('change', {target : { value : 'foo'}});
        expect(mockFn).toHaveBeenCalledTimes(1)
    })
})