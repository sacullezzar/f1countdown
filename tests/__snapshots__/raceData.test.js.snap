// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<raceData /> first test 1`] = `
<div
  className="input-group w-50 m-auto"
>
  <div
    className="input-group-prepend"
  >
    <label
      className="input-group-text"
      htmlFor="inputGroupSelect01"
    >
      Races
    </label>
  </div>
  <select
    className="custom-select"
    onChange={[Function]}
  >
    <option
      className="dropdown-item"
      key="1"
      value="2019-03-17T05:10:00Z"
    >
      Australian Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="2"
      value="2019-03-31T15:10:00Z"
    >
      Bahrain Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="3"
      value="2019-04-14T06:10:00Z"
    >
      Chinese Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="4"
      value="2019-04-28T12:10:00Z"
    >
      Azerbaijan Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="5"
      value="2019-05-12T13:10:00Z"
    >
      Spanish Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="6"
      value="2019-05-26T13:10:00Z"
    >
      Monaco Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="7"
      value="2019-06-09T18:10:00Z"
    >
      Canadian Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="8"
      value="2019-06-23T13:10:00Z"
    >
      French Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="9"
      value="2019-06-30T13:10:00Z"
    >
      Austrian Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="10"
      value="2019-07-14T13:10:00Z"
    >
      British Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="11"
      value="2019-07-28T13:10:00Z"
    >
      German Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="12"
      value="2019-08-04T13:10:00Z"
    >
      Hungarian Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="13"
      value="2019-09-01T13:10:00Z"
    >
      Belgian Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="14"
      value="2019-09-08T13:10:00Z"
    >
      Italian Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="15"
      value="2019-09-22T12:10:00Z"
    >
      Singapore Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="16"
      value="2019-09-29T11:10:00Z"
    >
      Russian Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="17"
      value="2019-10-13T05:10:00Z"
    >
      Japanese Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="18"
      value="2019-10-27T19:10:00Z"
    >
      Mexican Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="19"
      value="2019-11-03T19:10:00Z"
    >
      United States Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="20"
      value="2019-11-17T17:10:00Z"
    >
      Brazilian Grand Prix
    </option>
    <option
      className="dropdown-item"
      key="21"
      value="2019-12-01T13:10:00Z"
    >
      Abu Dhabi Grand Prix
    </option>
  </select>
</div>
`;
